Winleafs.Wpf/Views/Layout/CreateEffectWindow.xaml.cs

```diff
@@ -0,0 +1,262 @@
+			{
+				FrameListBox.SelectedIndex--;
+			}
+
+			_frameListItems.Remove(lastFrameListItem);
+			_customEffect.Frames.Remove(lastFrameListItem.Frame);
+		}
+
+		private void FrameListBox_SelectionChanged(object sender, System.Windows.Controls.SelectionChangedEventArgs e)
+		{
+			var selectedFrame = (FrameListBox.SelectedItem as FrameListItem)?.Frame;
+			if (selectedFrame == null)
+			{
+				return;
+			}
+
+			_currentFrame = selectedFrame;
+			
+			// Create a map of panels to the brushes to color them
+			var panelToBrushMap = new Dictionary<int, SolidColorBrush>();
+
+			foreach (var panel in _currentFrame.PanelColors)
+			{
+				panelToBrushMap.Add(panel.Key, _rgbToBrushMap[panel.Value]);
+			}
+
+			//Update the panels with the map
+			LayoutDisplay.PanelToBrushMap = panelToBrushMap;
+			LayoutDisplay.UpdateColors();
+		}
+
+		private void AddNewFrame()
+		{ 
+			//Copy the previous frame's panel colors where we can
+			var prevFrame = _customEffect.Frames.LastOrDefault();
+            var newFrame = new Frame();
+
+            foreach (var panelId in LayoutDisplay.PanelIds)
+            {
+				uint color = 0;
+				if (prevFrame != null && prevFrame.PanelColors.TryGetValue(panelId, out var prevColor))
+				{
+					color = prevColor;
+				}
+
+				newFrame.PanelColors.Add(panelId, color);
+            }                
+
+			// Add the frame to the effect and the displayed list and select it
+            _customEffect.Frames.Add(newFrame);
+			var newFrameListItem = new FrameListItem(newFrame, string.Format(_frameNameFormat, _frameListItems.Count + 1));
+			_frameListItems.Add(newFrameListItem);
+			FrameListBox.SelectedItem = newFrameListItem;
+        }
+
+		private void BuildFrameList()
+		{
+			_frameListItems = new ObservableCollection<FrameListItem>();
+
+			for (var i = 0; i < _customEffect.Frames.Count; i++)
+			{
+				_frameListItems.Add(new FrameListItem(_customEffect.Frames[i], string.Format(_frameNameFormat, i + 1)));
+			}
+
+			FrameListBox.ItemsSource = _frameListItems;
+
+			//Always show at least 1 frame in the list
+			if (_customEffect.Frames.Count == 0)
+			{
+				AddNewFrame();
+			}
+
+			FrameListBox.SelectedItem = _frameListItems.First();
+		}
+
+		private void BuildRgbToBrushMap()
+		{
+			//Get the colours used across all teh frames, and make sure black (not lit) is included
+			var colorsUsed = _customEffect.Frames.SelectMany(f => f.PanelColors.Values).Append(0U).Distinct();
+
+			//Add the colors used to the color picker control
+			ColorPicker.StandardColors.Clear();
+			_rgbToBrushMap = new Dictionary<uint, SolidColorBrush>();
+			
+			//Create a map of color to brush so that brushes can be reused
+			foreach (var rgb in colorsUsed)
+			{
+				AddToRgbToBrushMap(ColorFormatConverter.ToMediaColor(rgb));
+			}
+		}
+
+		private void AddToRgbToBrushMap(Color color)
+		{
+			ColorPicker.StandardColors.Add(new Xceed.Wpf.Toolkit.ColorItem(color, string.Empty));
```

Comment:
Could you use a using instead of the full namespace?

Reply:
Will do